confirm-deposit.md

```diff
@@ -0,0 +1,120 @@
+# Confirm Deposits to the Exchange
+
+Deposits are made in several stages. 
+After the user deposits funds to their assigned reception wallet (step 1 in [Detecting Deposits](https://docs.flare.network/exchange/architecture/#detecting-deposits)), it goes through the [Exchange server](https://docs.flare.network/exchange/architecture/#architecture-of-an-exchange), which monitors for deposits and updates the Balances database. 
+Run the code below to have the Exchange server listen for deposit transactions and confirm the deposits.
+
+On a [blockchain](https://docs.flare.network/tech/glossary/), multiple [validators](https://docs.flare.network/tech/validators/) examine newly created transactions and decide on whether they are valid and which [block](https://docs.flare.network/tech/glossary/) to add to the chain next. 
+It takes time for all the validators to agree.
+Until they reach agreement, validators may add or revert transactions to a block. 
+After the validation process is complete, reversing a transaction is very unlikely. 
```

Comment:
I see from the page on validators that probabilities of being a leader are involved in the consensus algorithm. Is there a point at which transactions on a block become irreversible/immutable?

Reply:
The description of the consensus you are giving here is generic, and I think that is OK. I don't think there's any need to go into the details of how the consensus works for this page. So no need to talk about leaders.

Regarding immutability of transactions, some blockchains implement "Finality", which is a process that guarantees that a transaction won't change. But not us (nor Ethereum, for instance), so there is no such guarantee.
ANY transaction can be reverted in the future, but the probability of that happening is reduced with each new block. Therefore after a certain number of blocks it is commonly accepted that the transaction is "secure". The exact number of blocks depends on the blockchain and the security you want to provide: it's a tradeoff between security and speed.
The example uses 5 blocks, I think Bitcoin and Ethereum use 6 blocks, and some Exchanges might use 12 blocks, just to be extra secure.

Reply:
Thanks.

Just to make sure (for my own understanding....no change in text), it sounds like you're saying that validators can revert a transaction, but it is understood that blocks are still tamper-resistant from malicious actors (by virtue of cryptography).